File: src/java/cache/CacheElement.java
package cache;

public class CacheElement<K, V> {

    /**
     * Cache element value.
     */
    private V mValue;

    /**
     * Cache element key.
     */
    private K mKey;

    /**
     * Counter for LFU algorithm.
     */
    private int mHitCount;

    public V getValue() {
        return mValue;
    }

    public void setValue(V value) {
        mValue = value;
    }

    public K getKey() {
        return mKey;
    }

    public void setKey(K key) {
        mKey = key;
    }

    public int getHitCount() {
        return mHitCount;
    }

    public void incrementHitCount() {
        mHitCount++;
    }

    public void setHitCount(int hitCount) {
        mHitCount = hitCount;
    }
}

File: src/com/nvv/cache/kotlin/CacheElement.kt
package com.nvv.cache.kotlin

public class CacheElement<K, V> {

    /**
    * Cache element value.
    */
    var value: V? = null

    /**
    * Cache element key.
    */
    var key: K? = null

    /**
    * Counter for LFU algorithm.
    */
    var hitCount : Int = 0

}